bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingTypeResource.java

```diff
@@ -184,9 +184,6 @@ public Response getByUID(@PathParam("thingTypeUID") @ApiParam(value = "thingType
 
         final List<ChannelDefinitionDTO> channelDefinitions = convertToChannelDefinitionDTOs(
                 thingType.getChannelDefinitions(), locale);
-        if (channelDefinitions == null) {
-            return null;
```

Comment:
Since it won't return `null` now `@Nullable` should be removed from the method result type.